TripKitUI/controller/TKSectionedAlertViewModel.swift

```diff
@@ -0,0 +1,138 @@
+//
+//  TKSectionedAlertViewModel.swift
+//  TripKitUI-iOS
+//
+//  Created by Kuan Lun Huang on 15/3/18.
+//  Copyright © 2018 SkedGo. All rights reserved.
+//
+
+import Foundation
+import RxSwift
+import RxDataSources
+
+public class TKSectionedAlertViewModel {
+  
+  public let region: SVKRegion
+  
+  private let disposeBag = DisposeBag()
+  
+  lazy var sections: Observable<[AlertSection]> = { [unowned self] in 
+    return TKBuzzInfoProvider.rx_fetchTransitAlertMappings(forRegion: region)
+      .map { self.groupAlertMappings($0) }
+      .map { self.alertSections(from: $0) }
+      .share(replay: 1)
+  }()
+  
+  public init(region: SVKRegion) {
+    self.region = region
+  }
+  
+  // MARK: -
+  
+  private func groupAlertMappings(_ mappings: [API.AlertMapping]) -> [String: [AlertGroup]] {
+    var alertGroupsByModes: [String: [AlertGroup]] = [:]
+    
+    for mapping in mappings {
+      mapping.routes?.forEach {
+        // TODO: Prefer to just use identifier.
+        let mode = $0.modeInfo.identifier ?? $0.modeInfo.alt
+        if var existingGroups = alertGroupsByModes[mode] {
+          let existingIds  = existingGroups.map { $0.route.id }
+          if let index = existingIds.index(of: $0.id) {
+            var currentAlerts = existingGroups[index].alerts
+            currentAlerts.append(mapping.alert)
+            existingGroups[index].alerts = currentAlerts
+            alertGroupsByModes[mode] = existingGroups
+          } else {
+            let newGroup = AlertGroup(route: $0, transportType: mapping.transportType, alerts: [mapping.alert])
+            existingGroups.append(newGroup)
+            alertGroupsByModes[mode] = existingGroups
+          }
+        } else {
+          let newGroup = AlertGroup(route: $0, transportType: mapping.transportType, alerts: [mapping.alert])
+          alertGroupsByModes[mode] = [newGroup]
+        }
+      }
+    }
```

Comment:
Left-over TODO and I still can't make sense of what this does. Please add a unit test for this.

Reply:
By the way, there's a new `Dictionary(grouping:by:)` which you might want to use instead: https://developer.apple.com/documentation/swift/dictionary/2919592-init